Makefile

```diff
@@ -38,52 +37,66 @@ ALL = \
 	\
 	$(ISO_ROOT)/pillar/repositories.sls \
 	$(ISO_ROOT)/pillar/top.sls \
+	\
+	$(ISO_ROOT)/product.txt \
 
 
 default: all
 .PHONY: default
 
-all: iso
+all: $(ISO)
 .PHONY: all
 
 all-local: $(ALL) ## Build all artifacts in the build tree
 .PHONY: all-local
 
-$(ISO_ROOT)/bootstrap.sh: scripts/bootstrap.sh.in
+$(ISO_ROOT)/bootstrap.sh: scripts/bootstrap.sh.in $(ISO_ROOT)/product.txt
 	mkdir -p $(shell dirname $@)
 	rm -f $@
-	sed s/@VERSION@/$(VERSION)/g < $< > $@ || rm -f $@
+	sed s/@VERSION@/$(shell source $(ISO_ROOT)/product.txt && echo $$SHORT_VERSION)/g < $< > $@ || (rm -f $@; false)
 	chmod a+x $@
 
 $(ISO_ROOT)/salt/%: salt/%
 	mkdir -p $(shell dirname $@)
 	rm -f $@
 	cp -a $< $@
 
-$(ISO_ROOT)/pillar/top.sls: pillar/top.sls.in
+$(ISO_ROOT)/pillar/top.sls: pillar/top.sls.in $(ISO_ROOT)/product.txt
 	mkdir -p $(shell dirname $@)
 	rm -f $@
-	sed s/@VERSION@/$(VERSION)/g < $< > $@ || rm -f $@
+	sed s/@VERSION@/$(shell source $(ISO_ROOT)/product.txt && echo $$SHORT_VERSION)/g < $< > $@ || (rm -f $@; false)
 
 $(ISO_ROOT)/pillar/%: pillar/%
 	mkdir -p $(shell dirname $@)
 	rm -f $@
 	cp -a $< $@
 
+$(ISO_ROOT)/product.txt: scripts/product.sh VERSION FORCE
+	rm -f $@
+	mkdir -p $(shell dirname $@)
+	env \
+		VERSION_MAJOR=$(VERSION_MAJOR) \
+		VERSION_MINOR=$(VERSION_MINOR) \
+		VERSION_PATCH=$(VERSION_PATCH) \
+		VERSION_SUFFIX=$(VERSION_SUFFIX) \
+		$< > $@ || (rm -f $@; false)
+
+FORCE:
+
 clean: ## Clean the build tree
 	rm -rf $(BUILD_ROOT)
 .PHONY: clean
 
 iso: $(ISO) ## Build the MetalK8s ISO image
 .PHONY: iso
 
-$(ISO): all-local
+$(ISO): $(ALL)
 	mkisofs -output $@ \
 		-rock \
 		-joliet \
 		-joliet-long \
 		-full-iso9660-filenames \
-		-volid 'MetalK8s $(FULL_VERSION)' \
+		-volid '$(shell source $(ISO_ROOT)/product.txt && echo $$NAME) $(shell source $(ISO_ROOT)/product.txt && echo $$VERSION)' \
```

Comment:
Can't the `source` be done only once in the rule (before the call to `mkisofs`, instead of each time for each command.

Or maybe even have a dedicated rule to set the env (since it's already used 4 times).

Reply:
> Or maybe even have a dedicated rule to set the env (since it's already used 4 times).

How would that work?

Reply:
Don't know if it would work, but a maybe Make rule that would do a bunch of `export` (or source a script with a bunch of `export`) of the necessary environment variable.

But if it doesn't work, then we can just `source` the file in each rule that needs it.

Reply:
Every command in a recipe is executed in a different subshell, so that wouldn't work as-is. Also, the file is generated by this `Makefile`, so the only way to somehow use the variables in a `Makefile` is to re-spawn `make`, which is... not great.

Changing the generated file to do a bunch of `export`s I don't really like because it makes parsing harder, and allows for people to start doing tricks like variable expansion and whatnot, expect it to work, but again makes parsing harder (if not impossible).

I'd like to keep things as-is for now, when it becomes really impractical we can reconsider.

Reply:
Ok for me, the change you already made is good enough for me.